### PR TITLE
feat: make ProcessBuildEvents Lambda timeout configurable

### PR DESCRIPTION
**Pull request by navinsoni**

### CODE REVIEW

File: template.yml [hunk 1]
```diff
@@ -59,6 +59,12 @@ Parameters:
       - "false"
     Default: "true"
     Description: Set to "false" to not publish a comment when build is successful.
+  BuildEventTimeout:
+    Type: Number
+    MinValue: 1
+    MaxValue: 900
+    Default: 60
+    Description: Timeout for Process Build Event Lambda.
```
Comment (jlhood) [inline template.yml:67]: Please update the README to document the new parameter.

**Review by jlhood — CHANGES_REQUESTED** [review]
Really appreciate this PR! Changes look good. Just need one minor README update, then it should be good to go!

**Review by jlhood — APPROVED** [review]

**Comment by jlhood** [comment]
The CI build failed for this PR with this error. Not sure if it's related to the dependency upgrades? Did you run into this problem locally running make?

```
test/unit/test_s3link.py::test_get_presigned_url_none_key PASSED         [ 88%]
--
231 | test/unit/test_s3link.py::test_get_presigned_url_key_not_found PASSED    [ 92%]
232 | test/unit/test_s3link.py::test_get_presigned_url_other_exception PASSED  [ 96%]
233 | test/unit/test_s3link.py::test_get_presigned_url_key_exists PASSED       [100%]
234 | INTERNALERROR> Traceback (most recent call last):
235 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 1107, in execute
236 | INTERNALERROR>     return self.con.execute(sql, parameters)
237 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/ext/sqlite3/patch.py", line 32, in execute
238 | INTERNALERROR>     return self.cursor().execute(*args, **kwargs)
239 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/models/subsegment.py", line 58, in __call__
240 | INTERNALERROR>     meta_processor=None,
241 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 428, in record_subsegment
242 | INTERNALERROR>     subsegment = self.begin_subsegment(name, namespace)
243 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 292, in begin_subsegment
244 | INTERNALERROR>     segment = self.current_segment()
245 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 270, in current_segment
246 | INTERNALERROR>     entity = self.get_trace_entity()
247 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 390, in get_trace_entity
248 | INTERNALERROR>     return self.context.get_trace_entity()
249 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/context.py", line 99, in get_trace_entity
250 | INTERNALERROR>     return self.handle_context_missing()
251 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/context.py", line 123, in handle_context_missing
252 | INTERNALERROR>     raise SegmentNotFoundException(MISSING_SEGMENT_MSG)
253 | INTERNALERROR> aws_xray_sdk.core.exceptions.exceptions.SegmentNotFoundException: cannot find the current segment/subsegment, please make sure you have a segment open
254 | INTERNALERROR>
255 | INTERNALERROR> During handling of the above exception, another exception occurred:
256 | INTERNALERROR>
257 | INTERNALERROR> Traceback (most recent call last):
258 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/_pytest/main.py", line 268, in wrap_session
259 | INTERNALERROR>     session.exitstatus = doit(config, session) or 0
260 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/_pytest/main.py", line 322, in _main
261 | INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
262 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pluggy/_hooks.py", line 265, in __call__
263 | INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
264 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pluggy/_manager.py", line 80, in _hookexec
265 | INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
266 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pluggy/_callers.py", line 55, in _multicall
267 | INTERNALERROR>     gen.send(outcome)
268 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pytest_cov/plugin.py", line 294, in pytest_runtestloop
269 | INTERNALERROR>     self.cov_controller.finish()
270 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pytest_cov/engine.py", line 44, in ensure_topdir_wrapper
271 | INTERNALERROR>     return meth(self, *args, **kwargs)
272 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pytest_cov/engine.py", line 230, in finish
273 | INTERNALERROR>     self.cov.stop()
274 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/control.py", line 705, in save
275 | INTERNALERROR>     data = self.get_data()
276 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/control.py", line 774, in get_data
277 | INTERNALERROR>     if self._collector and self._collector.flush_data():
278 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/collector.py", line 473, in flush_data
279 | INTERNALERROR>     self.covdata.add_lines(self.mapped_file_dict(self.data))
280 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 240, in _wrapped
281 | INTERNALERROR>     return method(self, *args, **kwargs)
282 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 463, in add_lines
283 | INTERNALERROR>     self._choose_lines_or_arcs(lines=True)
284 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 529, in _choose_lines_or_arcs
285 | INTERNALERROR>     with self._connect() as con:
286 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 317, in _connect
287 | INTERNALERROR>     self._open_db()
288 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 268, in _open_db
289 | INTERNALERROR>     self._read_db()
290 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 272, in _read_db
291 | INTERNALERROR>     with self._dbs[threading.get_ident()] as db:
292 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 1084, in __enter__
293 | INTERNALERROR>     self._connect()
294 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 1072, in _connect
295 | INTERNALERROR>     self.execute("pragma journal_mode=off").close()
296 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 1112, in execute
297 | INTERNALERROR>     return self.con.execute(sql, parameters)
298 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/ext/sqlite3/patch.py", line 32, in execute
299 | INTERNALERROR>     return self.cursor().execute(*args, **kwargs)
300 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/models/subsegment.py", line 58, in __call__
301 | INTERNALERROR>     meta_processor=None,
302 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 428, in record_subsegment
303 | INTERNALERROR>     subsegment = self.begin_subsegment(name, namespace)
304 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 292, in begin_subsegment
305 | INTERNALERROR>     segment = self.current_segment()
306 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 270, in current_segment
307 | INTERNALERROR>     entity = self.get_trace_entity()
308 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 390, in get_trace_entity
309 | INTERNALERROR>     return self.context.get_trace_entity()
310 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/context.py", line 99, in get_trace_entity
311 | INTERNALERROR>     return self.handle_context_missing()
312 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/context.py", line 123, in handle_context_missing
313 | INTERNALERROR>     raise SegmentNotFoundException(MISSING_SEGMENT_MSG)
314 | INTERNALERROR> aws_xray_sdk.core.exceptions.exceptions.SegmentNotFoundException: cannot find the current segment/subsegment, please make sure you have a segment open
315 |  
316 | ============================== 26 passed in 0.81s ==============================
317 | make: *** [compile] Error 3
```

**Comment by navinsoni** [comment]
Yes I saw this error locally as well 

> The CI build failed for this PR with this error. Not sure if it's related to the dependency upgrades? Did you run into this problem locally running make?
> 
> 
> 
> ```
> 
> test/unit/test_s3link.py::test_get_presigned_url_none_key PASSED         [ 88%]
> 
> --
> 
> 231 | test/unit/test_s3link.py::test_get_presigned_url_key_not_found PASSED    [ 92%]
> 
> 232 | test/unit/test_s3link.py::test_get_presigned_url_other_exception PASSED  [ 96%]
> 
> 233 | test/unit/test_s3link.py::test_get_presigned_url_key_exists PASSED       [100%]
> 
> 234 | INTERNALERROR> Traceback (most recent call last):
> 
> 235 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 1107, in execute
> 
> 236 | INTERNALERROR>     return self.con.execute(sql, parameters)
> 
> 237 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/ext/sqlite3/patch.py", line 32, in execute
> 
> 238 | INTERNALERROR>     return self.cursor().execute(*args, **kwargs)
> 
> 239 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/models/subsegment.py", line 58, in __call__
> 
> 240 | INTERNALERROR>     meta_processor=None,
> 
> 241 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 428, in record_subsegment
> 
> 242 | INTERNALERROR>     subsegment = self.begin_subsegment(name, namespace)
> 
> 243 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 292, in begin_subsegment
> 
> 244 | INTERNALERROR>     segment = self.current_segment()
> 
> 245 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 270, in current_segment
> 
> 246 | INTERNALERROR>     entity = self.get_trace_entity()
> 
> 247 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 390, in get_trace_entity
> 
> 248 | INTERNALERROR>     return self.context.get_trace_entity()
> 
> 249 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/context.py", line 99, in get_trace_entity
> 
> 250 | INTERNALERROR>     return self.handle_context_missing()
> 
> 251 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/context.py", line 123, in handle_context_missing
> 
> 252 | INTERNALERROR>     raise SegmentNotFoundException(MISSING_SEGMENT_MSG)
> 
> 253 | INTERNALERROR> aws_xray_sdk.core.exceptions.exceptions.SegmentNotFoundException: cannot find the current segment/subsegment, please make sure you have a segment open
> 
> 254 | INTERNALERROR>
> 
> 255 | INTERNALERROR> During handling of the above exception, another exception occurred:
> 
> 256 | INTERNALERROR>
> 
> 257 | INTERNALERROR> Traceback (most recent call last):
> 
> 258 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/_pytest/main.py", line 268, in wrap_session
> 
> 259 | INTERNALERROR>     session.exitstatus = doit(config, session) or 0
> 
> 260 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/_pytest/main.py", line 322, in _main
> 
> 261 | INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
> 
> 262 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pluggy/_hooks.py", line 265, in __call__
> 
> 263 | INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
> 
> 264 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pluggy/_manager.py", line 80, in _hookexec
> 
> 265 | INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
> 
> 266 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pluggy/_callers.py", line 55, in _multicall
> 
> 267 | INTERNALERROR>     gen.send(outcome)
> 
> 268 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pytest_cov/plugin.py", line 294, in pytest_runtestloop
> 
> 269 | INTERNALERROR>     self.cov_controller.finish()
> 
> 270 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pytest_cov/engine.py", line 44, in ensure_topdir_wrapper
> 
> 271 | INTERNALERROR>     return meth(self, *args, **kwargs)
> 
> 272 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/pytest_cov/engine.py", line 230, in finish
> 
> 273 | INTERNALERROR>     self.cov.stop()
> 
> 274 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/control.py", line 705, in save
> 
> 275 | INTERNALERROR>     data = self.get_data()
> 
> 276 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/control.py", line 774, in get_data
> 
> 277 | INTERNALERROR>     if self._collector and self._collector.flush_data():
> 
> 278 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/collector.py", line 473, in flush_data
> 
> 279 | INTERNALERROR>     self.covdata.add_lines(self.mapped_file_dict(self.data))
> 
> 280 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 240, in _wrapped
> 
> 281 | INTERNALERROR>     return method(self, *args, **kwargs)
> 
> 282 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 463, in add_lines
> 
> 283 | INTERNALERROR>     self._choose_lines_or_arcs(lines=True)
> 
> 284 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 529, in _choose_lines_or_arcs
> 
> 285 | INTERNALERROR>     with self._connect() as con:
> 
> 286 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 317, in _connect
> 
> 287 | INTERNALERROR>     self._open_db()
> 
> 288 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 268, in _open_db
> 
> 289 | INTERNALERROR>     self._read_db()
> 
> 290 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 272, in _read_db
> 
> 291 | INTERNALERROR>     with self._dbs[threading.get_ident()] as db:
> 
> 292 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 1084, in __enter__
> 
> 293 | INTERNALERROR>     self._connect()
> 
> 294 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 1072, in _connect
> 
> 295 | INTERNALERROR>     self.execute("pragma journal_mode=off").close()
> 
> 296 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/coverage/sqldata.py", line 1112, in execute
> 
> 297 | INTERNALERROR>     return self.con.execute(sql, parameters)
> 
> 298 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/ext/sqlite3/patch.py", line 32, in execute
> 
> 299 | INTERNALERROR>     return self.cursor().execute(*args, **kwargs)
> 
> 300 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/models/subsegment.py", line 58, in __call__
> 
> 301 | INTERNALERROR>     meta_processor=None,
> 
> 302 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 428, in record_subsegment
> 
> 303 | INTERNALERROR>     subsegment = self.begin_subsegment(name, namespace)
> 
> 304 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 292, in begin_subsegment
> 
> 305 | INTERNALERROR>     segment = self.current_segment()
> 
> 306 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 270, in current_segment
> 
> 307 | INTERNALERROR>     entity = self.get_trace_entity()
> 
> 308 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/recorder.py", line 390, in get_trace_entity
> 
> 309 | INTERNALERROR>     return self.context.get_trace_entity()
> 
> 310 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/context.py", line 99, in get_trace_entity
> 
> 311 | INTERNALERROR>     return self.handle_context_missing()
> 
> 312 | INTERNALERROR>   File "/root/.local/share/virtualenvs/github-codebuild-logs-O1BByKoz/lib/python3.7/site-packages/aws_xray_sdk/core/context.py", line 123, in handle_context_missing
> 
> 313 | INTERNALERROR>     raise SegmentNotFoundException(MISSING_SEGMENT_MSG)
> 
> 314 | INTERNALERROR> aws_xray_sdk.core.exceptions.exceptions.SegmentNotFoundException: cannot find the current segment/subsegment, please make sure you have a segment open
> 
> 315 |  
> 
> 316 | ============================== 26 passed in 0.81s ==============================
> 
> 317 | make: *** [compile] Error 3
> 
> ```

**Comment by jlhood** [comment]
CI build is still failing, but it's due to an issue with the CI setup (really old setup, needs to be updated). The unit tests all pass now with the latest changes. Going to go ahead and merge this PR.

File: Makefile [hunk 2]
```diff
@@ -43,7 +43,7 @@ compile:
 	pipenv run cfn-lint template.yml
 	pipenv run py.test --cov=$(SRC_DIR) --cov-fail-under=85 -l -vv test/unit
 	pipenv lock --requirements > $(SRC_DIR)/requirements.txt
-	sam build
+	sam build --use-container
```
Comment (jlhood) [inline Makefile:46]: This change is causing the CI build to fail with this error:

```
Build Failed
Error: Docker is unreachable. Docker needs to be running to build inside a container.
```

**Review by jlhood — CHANGES_REQUESTED** [review]

**Review by jlhood — APPROVED** [review]

**Review by jlhood — APPROVED** [review]

**Pull request merged**